File: 02_EstructurasControl/Repetitivas/While/C#/Ejermplo-01/Ejemplo01.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _1.While_tarearepositorio3384
{
    internal class Program
    {
        static void Main(string[] args)
        {
            //Simula la descarga de un archivo, mostrando el progreso hasta que se complete.
            int progreso = 0;
            Console.WriteLine("Descargando archivo...");
            while (progreso < 100)
            {
                progreso += 10; // Incrementa el progreso en 10%
                if (progreso > 100) progreso = 100; // Asegura que no exceda 100%
                Console.WriteLine($"Progreso: {progreso}%");
                System.Threading.Thread.Sleep(500); // Simula tiempo de descarga
            }
            Console.Write("Descarga completa.");
            Console.ReadLine();
        }
    }
}

File: 02_EstructurasControl/Repetitivas/While/Python/Ejemplo01.py
import time  # Para simular el tiempo de descarga

# Simula la descarga de un archivo, mostrando el progreso hasta que se complete
progreso = 0
print("Descargando archivo...")

while progreso < 100:
    progreso += 10  # Incrementa el progreso en 10%
    if progreso > 100:
        progreso = 100  # Asegura que no exceda 100%
    print(f"Progreso: {progreso}%")
    time.sleep(0.5)  # Simula tiempo de descarga (500 ms)

print("Descarga completa.")
input("Presione Enter para continuar...")  # Pausa para ver el resultado
